src/MongoDB/Monitoring/ServerOpeningEvent.c

```diff
@@ -0,0 +1,141 @@
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_serveropeningevent_ce;
+
+/* {{{ proto string ServerOpeningEvent::getHost()
+   Returns this event's host */
+static PHP_METHOD(ServerOpeningEvent, getHost)
+{
+	php_phongo_serveropeningevent_t* intern = Z_SERVEROPENINGEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	RETVAL_STRING((intern->host)->host);
+} /* }}} */
+
+/* {{{ proto MongoDB\BSON\ObjectId ServerOpeningEvent::getTopologyId()
+   Returns this event's topology id */
+static PHP_METHOD(ServerOpeningEvent, getTopologyId)
+{
+	php_phongo_objectid_t*           topology_id;
+	php_phongo_serveropeningevent_t* intern = Z_SERVEROPENINGEVENT_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	object_init_ex(return_value, php_phongo_objectid_ce);
+
+	topology_id = Z_OBJECTID_OBJ_P(return_value);
+	bson_oid_to_string(&intern->topology_id, topology_id->oid);
+	topology_id->initialized = true;
+} /* }}} */
+
+/* {{{ MongoDB\Driver\Monitoring\ServerOpeningEvent function entries */
+ZEND_BEGIN_ARG_INFO_EX(ai_ServerOpeningEvent_void, 0, 0, 0)
+ZEND_END_ARG_INFO()
+
+static zend_function_entry php_phongo_serveropeningevent_me[] = {
+	/* clang-format off */
+	ZEND_NAMED_ME(__construct, PHP_FN(MongoDB_disabled___construct), ai_ServerOpeningEvent_void, ZEND_ACC_PRIVATE | ZEND_ACC_FINAL)
+	PHP_ME(ServerOpeningEvent, getHost, ai_ServerOpeningEvent_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_ME(ServerOpeningEvent, getTopologyId, ai_ServerOpeningEvent_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	ZEND_NAMED_ME(__wakeup, PHP_FN(MongoDB_disabled___wakeup), ai_ServerOpeningEvent_void, ZEND_ACC_PUBLIC | ZEND_ACC_FINAL)
+	PHP_FE_END
+	/* clang-format on */
+};
+/* }}} */
+
+/* {{{ MongoDB\Driver\Monitoring\ServerOpeningEvent object handlers */
+static zend_object_handlers php_phongo_handler_serveropeningevent;
+
+static void php_phongo_serveropeningevent_free_object(zend_object* object) /* {{{ */
+{
+	php_phongo_serveropeningevent_t* intern = Z_OBJ_SERVEROPENINGEVENT(object);
+
+	zend_object_std_dtor(&intern->std);
+} /* }}} */
+
+static zend_object* php_phongo_serveropeningevent_create_object(zend_class_entry* class_type) /* {{{ */
+{
+	php_phongo_serveropeningevent_t* intern = NULL;
+
+	intern = PHONGO_ALLOC_OBJECT_T(php_phongo_serveropeningevent_t, class_type);
+
+	zend_object_std_init(&intern->std, class_type);
+	object_properties_init(&intern->std, class_type);
+
+	intern->std.handlers = &php_phongo_handler_serveropeningevent;
+
+	return &intern->std;
+} /* }}} */
+
+static HashTable* php_phongo_serveropeningevent_get_debug_info(phongo_compat_object_handler_type* object, int* is_temp) /* {{{ */
+{
+	php_phongo_serveropeningevent_t* intern;
+	zval                             retval = ZVAL_STATIC_INIT;
+	char                             topology_id[25];
+
+	intern   = Z_OBJ_SERVEROPENINGEVENT(PHONGO_COMPAT_GET_OBJ(object));
+	*is_temp = 1;
+	array_init_size(&retval, 2);
+
+	ADD_ASSOC_STRING(&retval, "host", (intern->host)->host);
```

Comment:
Note that this line will change depending on how you restructure `php_phongo_serveropeningevent_t`. At the very least, you'll definitely need to add a `port` field.